consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/TransitionProtocolSchedule.java

```diff
@@ -112,12 +99,8 @@ public ProtocolSchedule getPostMergeSchedule() {
    */
   @Override
   public ProtocolSpec getByBlockHeader(final ProcessableBlockHeader blockHeader) {
-    return this.timestampSchedule
-        .getByTimestamp(blockHeader.getTimestamp())
-        .orElseGet(
-            () ->
-                transitionUtils.dispatchFunctionAccordingToMergeState(
-                    protocolSchedule -> protocolSchedule.getByBlockHeader(blockHeader)));
```

Comment:
getByTimestamp and getByBlockNumber are stitched together inside UnifiedProtocolSchedule now, so complexity is encapsulated in that one place.